### PR TITLE
Master appraisal goal yti

### PR DESCRIPTION
**Pull request by tivisse**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/survey/controllers/main.py [hunk 1]
```diff
@@ -670,17 +684,7 @@ def _extract_filters_data(self, survey, post):
...
-            matching_line_ids = request.env['survey.user_input.line'].sudo().search(line_filter_domain).ids
-            user_input_domain = expression.AND([
-                [('user_input_line_ids', 'in', matching_line_ids)],
-                user_input_domain
-            ])
-        if post.get('finished'):
-            user_input_domain = expression.AND([[('state', '=', 'done')], user_input_domain])
-        else:
-            user_input_domain = expression.AND([[('state', '!=', 'new')], user_input_domain])
+        user_input_domain = self._get_user_input_domain(survey, line_filter_domain, post)
```
Comment (awa-odoo) [inline addons/survey/controllers/main.py:687]: Wondering if it would be cleaner to introduce a '_get_extra_user_input_domain' that would return False in survey.
So you don't split the domain building in multiple methods, which is a bit hard to follow.

So here you would have:
```py
extra_user_input_domain = self._get_extra_user_input_domain()
if extra_user_input_domain:
    user_input_domain = expression.AND(extra_user_input_domain, user_input_domain)

...

def _get_extra_user_input_domain(self, **post):
    return False
```

And in your appraisal you would simply add the necessary domain.
That simplifies parameters as well, you don't have to pass extra stuff (that you don't use at all in appraisal).

What do you think?

**Comment by tivisse** [comment]
@robodoo r+
